game/templates/game/opening_detail.html

```
@@ -0,0 +1,67 @@
+
+    <div class="progress-box">
+        <div id="move-progress">
+            0 / {{ opening.moves|length }}
+        </div>
+        <input
+            type="text"
+            id="move-input"
+            placeholder="Enter move (e4, Nf3...)"
+        />
+
+        <button id="check-move-btn">
```

HTMLHint (1.9.2)

[warning] The type attribute must be present on <button> elements.

(button-type-require)